ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/SpacingInUnaryExpressionRule.kt

```diff
@@ -0,0 +1,32 @@
+package com.pinterest.ktlint.ruleset.standard
+
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType
+import com.pinterest.ktlint.core.ast.children
+import com.pinterest.ktlint.core.ast.isPartOfComment
+import com.pinterest.ktlint.core.ast.isWhiteSpace
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+
+class SpacingInUnaryExpressionRule : Rule("unary-op-spacing") {
+
+    override fun visit(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit
+    ) {
+        if (node.elementType == ElementType.PREFIX_EXPRESSION ||
+            node.elementType == ElementType.POSTFIX_EXPRESSION
+        ) {
+            val children = node.children().toList()
+
+            // ignore: var a = + /* comment */ 1
+            if (children.any { it.isPartOfComment() }) return
+
+            val whiteSpace = children.firstOrNull { it.isWhiteSpace() } ?: return
+            if (autoCorrect) {
```

Comment:
I think, I would put this block after `emit`, because otherwise in case of autoCorrect, the node does not contain the whitespace already and the message is gonna be misleading

Reply:
oops, nice catch! will fix it